docs/source/developer/repomap.rst

```diff
@@ -0,0 +1,447 @@
+- ``numba/lowering.py`` - General implementation of lowering Numba IR to LLVM
+- ``numba/withcontexts.py`` - General scaffolding for implementing context managers in nopython mode, and the objectmode context manager
+- ``numba/pylowering.py`` - Lowering of Numba IR in object mode
+- ``numba/pythonapi.py`` - LLVM IR code generation to interface with CPython API
+
+
+Type Management
+'''''''''''''''
+- ``numba/typeconv/`` - Implementation of type casting and type signature matching in both C++ and Python
+- ``numba/capsulethunk.h`` - Used by typeconv
+- ``numba/types/`` - definition of the Numba type hierarchy, used everywhere in compiler to select implementations
+- ``numba/consts.py`` - Constant inference (used to make constant values available during codegen when possible)
+- ``numba/datamodel`` - LLVM IR representations of data types in different contexts
+- ``numba/datamodel/models.py`` - Models for most standard types
+- ``numba/datamodel/registry.py`` - Decorator to register new data models
+- ``numba/datamodel/packer.py`` - Pack typed values into a data structure
+- ``numba/datamodel/testing.py`` - Data model tests (this should move??)
+- ``numba/datamodel/manager.py`` - Map types to data models
+
+
+Compiled Extensions
+'''''''''''''''''''
+
+Numba uses a small amount of compiled C/C++ code for core functionality, like
+dispatching and type matching where performance matters, and it is more
+convient to encapsulate direct interaction with CPython APIs.
+
+- ``numba/_arraystruct.h`` - Struct for holding NumPy array attributes.  Used in helperlib and nrt.
+- ``numba/_helperlib.c`` - C functions required by Numba compiled code at runtime.  Linked into ahead-of-time compiled modules
+- ``numba/_helpermod.c`` - Python extension module with pointers to functions from ``_helperlib.c`` and ``_npymath_exports.c``
+- ``numba/_npymath_exports.c`` - Export function pointer table to NumPy C math functions
+- ``numba/_dynfuncmod.c`` - Python extension module exporting _dynfunc.c functionality
+- ``numba/_dynfunc.c`` - C level Environment and Closure objects (keep in sync with numba/target/base.py)
+- ``numba/mathnames.h`` - Macros for defining names of math functions
+- ``numba/_pymodule.h`` - C macros for Python 2/3 portable naming of C API functions
+- ``numba/_math_c99.{h,c}`` - C99 math compatibility (needed Python 2.7 on Windows, compiled with VS2008)
+- ``numba/mviewbuf.c`` - Handles Python memorviews
+- ``numba/_typeof.{h,c}`` - C implementation of type fingerprinting, used by dispatcher
+- ``numba/_numba_common.h`` - Portable C macro for marking symbols that can be shared between object files, but not outside the library.
+
+
+
+Misc Support
+''''''''''''
+
+- ``numba/_version.py`` - Updated by versioneer
+- ``numba/runtime`` - Global singleton that manages memory allocated on the heap by Numba-compiled functions
```

Comment:
Let's call out that it's the "language runtime".  Though all it does now is memory and reference-counting.